### ISSUE TITLE
Build Issue - Android 27 licenses not accepted - Fix proposal

### ISSUE THREAD

**Issue by Catfriend1** [syncthing/syncthing-android#1026]
Hi,

I saw the builds are failing. But on my private build server the apk builds fine with Android SDK 27. By the way - can I also submit something like a pull request for the travis CI build script?
I found the following lines by googling:

before_install:
  - yes | sdkmanager "platforms;android-27"

Haven't tried them yet. Maybe someone from the devs can try?

Thanks for your help, I'm just new to github and wanna do more contributions in the near future.
Regards

**Comment by Catfriend1** [comment]
Ok, just verified by adding the above line to before install in .travis.yml and started a travis build. The license problem got away but a new error popped up building native syncthing:

github.com/syncthing/syncthing/vendor/github.com/vitrun/qart/qr
text/template/parse
runtime/trace
os/signal
text/template
html/template
net/http/pprof
exit status 2
exit status 1
:syncthing:buildNative FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':syncthing:buildNative'.
> Process 'command './build-syncthing.bash'' finished with non-zero exit value 1
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
* Get more help at https://help.gradle.org
BUILD FAILED in 39s
1 actionable task: 1 executed
The command "./gradlew buildNative assembleDebug" exited with 1.

**Comment by Nutomic** [comment]
You're right, I actually started to work on this two days ago on the [fix-travis](https://github.com/syncthing/syncthing-android/commits/fix-travis) branch.

@calmh @AudriusButkevicius Any idea what that error means? [Full log](https://travis-ci.org/syncthing/syncthing-android/builds/345178593)

**Comment by calmh** [comment]
You’re building with a very old Go version;

> Go version:   go1.7.5

Use at least 1.9.

**Comment by Nutomic** [comment]
That is the preinstalled version, but the script explicitly installs Go 1.9:

https://github.com/syncthing/syncthing-android/blob/7369e54e2a02f5708bc10c08db5f503ec1d010f5/.travis.yml#L26

**Comment by Nutomic** [comment]
Okay you're right, for some reason it's using Go 1.7, even though 1.9 is installed.

**Comment by Nutomic** [comment]
That problem is fixed, but now theres another error that I don't understand (something to do with the Android NDK).

```
# runtime/cgo
In file included from gcc_libinit.c:8:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/pthread.h:37:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/time.h:33:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/sys/time.h:37:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/sys/select.h:36:
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/signal.h:101:18: error: expected ';' at end of declaration list
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/asm/signal.h:101:22: note: expanded from macro 'sa_handler'
In file included from gcc_libinit.c:8:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/pthread.h:37:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/time.h:33:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/sys/time.h:37:
In file included from /home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/sys/select.h:36:
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/signal.h:102:12: error: expected ')'
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/asm/signal.h:102:24: note: expanded from macro 'sa_sigaction'
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/signal.h:102:10: note: to match this '('
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/signal.h:102:12: error: duplicate member '_u'
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/asm/signal.h:102:22: note: expanded from macro 'sa_sigaction'
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/signal.h:101:18: note: previous declaration is here
/home/travis/build/syncthing/syncthing-android/syncthing/gobuild/standalone-ndk/android-14-arm/bin/../sysroot/usr/include/asm/signal.h:101:20: note: expanded from macro 'sa_handler'
:syncthing:buildNative FAILED
```

https://travis-ci.org/syncthing/syncthing-android/builds/345268457

**Comment by Catfriend1** [comment]
I had this too on my private build server, fixed it by installing go 1.9 instead of 1.10
